### ISSUE TITLE
The float value would map to int in enum values e.g `1.0` -> `1`

### ISSUE THREAD

**Issue by MaryGao** [microsoft/typespec#2240]
See [the example](https://cadlplayground.z22.web.core.windows.net/?c=ZW51bSBJbnRlZ2VyRcQMew0KICBPbmU6IDEsxAtUd286IDLGC2hyZWU6IDMsDQp9DQoNCsU8RmxvYXTQOi4wyzzIDcc%2BxQ%2FFQG1vZGVsIFJlc3VsdMY%2BaW50OuwAjsU5ZsRiOsppxz5Ac2VydmljZSh7fSkNCm5hbWVzcGFjZSBUZXPHTm9wIGdldCgpOsdkOw0KfQ%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%7D) for `FloatEnum` which has values `1.0, 2.0, 3.0` would be mapped to `1, 2, 3`. I think this is not correct.

```
enum FloatEnum {
  One: 1.0,
  Two: 2.0,
  Three: 3.0,
}
```

**Comment by timotheeguerin** [comment]
Seems like we have a typo in the docs, the example above show the correct syntax with `:` instead of `=`

**Comment by MaryGao** [comment]
@timotheeguerin Does that mean every emitter needs to check by themselves if they have requirement to distinguish `1.0` and `1`?
```
model Result {
  prop: 1.0   // Here emitter may need to chech if the devired type is int or float
}
```

**Comment by timotheeguerin** [comment]
yeah.. which is not ideal. Or we need to have at least an helper of the sort `getNumericPrecision(numericType) // => int32, float64, etc.)`

**Comment by markcowl** [comment]
fix the above doc

**Comment by timotheeguerin** [comment]
Doc fixed in https://github.com/microsoft/typespec/pull/2245, this can remain as a design discussion

**Comment by ArcturusZhang** [comment]
I opened this issue: https://github.com/microsoft/typespec/issues/2240
This would be an issue in csharp because adding a new value will cause breaking changes in C#
for example, in one version we have:
```
enum Foo {
A: 1.0;
B: 2.0;
}
```
and everything will generate as integers. When they add a real float number, everything suddenly changed to generate as float:
```
enum Foo {
A: 1.0;
B: 2.0;
C: 3.3;
}
```
